feature_layers/feature-collection-layer-from-portal/src/main/java/com/esri/samples/feature_collection_layer_from_portal/FeatureCollectionLayerFromPortalSample.java

```diff
@@ -108,25 +101,34 @@ public void start(Stage stage) {
      * Fetch feature collection from portal and load as feature layer.
      */
     private void fetchFeatureCollectionFromPortal() {
-
-        // create portal and portal item
-        portal = new Portal("https://www.arcgis.com/");
-        PortalItem portalItem = new PortalItem(portal, input.getText());
-
-        // create feature collection and add to the map as a layer
-        featureCollection = new FeatureCollection(portalItem);
-        featureCollectionLayer = new FeatureCollectionLayer(featureCollection);
-
-        featureCollectionLayer.loadAsync();
-        featureCollectionLayer.addDoneLoadingListener(() -> {
-            if (portalItem.getType() != PortalItem.Type.FEATURE_COLLECTION) {
-                new Alert(Alert.AlertType.ERROR, "This is not valid Feature Collection.").show();
-            }
-
-        });
+        if (!inputTextField.getText().isEmpty()) {
+            // crate portal item
+            PortalItem portalItem = new PortalItem(portal, inputTextField.getText());
+                // create feature collection and add to the map as a layer
+                featureCollection = new FeatureCollection(portalItem);
+                FeatureCollectionLayer  featureCollectionLayer = new FeatureCollectionLayer(featureCollection);
+                // add feature layer to the map
+                map.getOperationalLayers().clear();
+                map.getOperationalLayers().add(featureCollectionLayer);
+
+                featureCollectionLayer.addDoneLoadingListener(() -> {
+                    if(portalItem.getLoadStatus()== LoadStatus.LOADED) {
+                        if (portalItem.getType() != PortalItem.Type.FEATURE_COLLECTION) {
+                            new Alert(Alert.AlertType.ERROR, "This is not valid Feature Collection.").show();
+                        }
+                    }
+                    else
+                        {
+                            new Alert(Alert.AlertType.ERROR, portalItem.getLoadError().getCause().toString()).show();
+                        }
+                });
+        }
+        else
+        {
+            new Alert(Alert.AlertType.ERROR, "Portal Item ID is empty. Please enter a Portal Item ID.").show();
+        }
 
     }
```

Comment:
Nice work on this! There's only a couple of things left to tidy here:
A bit of whitespace would help make the users read this a little easier. As a general rule of thumb, we try and add a blank line between each block of code.

For the sake of making an easy to follow sample, we'd want to do a few things and consider the order of doing them:
1. check that the portal item ID is provided
2. check that we can load the portal item, and that it loaded successfuly (Adding the FCL should only happen after we check that the portal item has done loading successfuly. That way we don't perform the action before the appropriate resource is loaded. )
3. check that the portal item is a feature collection (We need to check that the portal item is a feature collection before actually trying to use it to create the FCL )
4. create the feature collection layer
5. add the later to the map.

This would make it a lot easier for users to read the sample and understand in which order stuff needs to be done.

Keep in mind that we don't generally print out the loading error using `portalItem.getLoadError().getCause().toString()`, as you've done, because it will be displayed to they end-user, and they wouldn't know what to do with the technical information. Instead we just write a message that describes what went wrong.

```suggestion
    /**
     * Fetch a feature collection from a portal and load it as a feature layer.
     */
    private void fetchFeatureCollectionFromPortal() {
        if (!inputTextField.getText().isEmpty()) {
            // crate portal item
            PortalItem portalItem = new PortalItem(portal, inputTextField.getText());

            // load the portal item
            portalItem.loadAsync();
            portalItem.addDoneLoadingListener(() -> {
                if (portalItem.getLoadStatus() == LoadStatus.LOADED) {

                    // check that the portal item is a feature collection
                    if (portalItem.getType() == PortalItem.Type.FEATURE_COLLECTION) {

                        // create a feature collection layer
                        featureCollection = new FeatureCollection(portalItem);
                        FeatureCollectionLayer featureCollectionLayer = new FeatureCollectionLayer(featureCollection);

                        // add the feature layer to the map
                        map.getOperationalLayers().clear();
                        map.getOperationalLayers().add(featureCollectionLayer);

                    } else {
                        new Alert(Alert.AlertType.ERROR, "Portal Item is not a valid Feature Collection").show();
                    }
                } else {
                    new Alert(Alert.AlertType.ERROR, "Failed to load Portal Item.").show();
                }
            });
        } else {
            new Alert(Alert.AlertType.ERROR, "Portal Item ID is empty. Please enter a Portal Item ID.").show();
        }
    }
    
```